src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -0,0 +1,194 @@
+          case Failure(e) => Failure(new IllegalArgumentException(s"JES requires File outputs to be determined prior to running, but ${taskOutput.name} can not."))
+        }
+      case _ => Success(None)
+    }
+  }
+
+  def taskOutputsToJesOutputs(taskOutputs: Seq[TaskOutput], callGcsPath: String, scopedLookupFunction: ScopedLookupFunction, engineFunctions: JesEngineFunctions): Try[Seq[JesOutput]] = {
+    val localizedOutputs = taskOutputs map {taskOutputToJesOutput(_, callGcsPath, scopedLookupFunction, engineFunctions)}
+    val localizationFailures = localizedOutputs filter {_.isFailure}
+    if (localizationFailures.isEmpty) {
+      Success(localizedOutputs.collect({case Success(x) => x}).flatten)
+    } else {
+      val failureMessages = TryUtil.stringifyFailures(localizationFailures)
+      Failure(new IllegalArgumentException(failureMessages.mkString("\n")))
+    }
+  }
+
+  def anonymousTaskOutput(value: String, engineFunctions: JesEngineFunctions): JesOutput = {
+    JesOutput(value, engineFunctions.gcsPathFromAnyString(value).toString, Paths.get(value))
+  }
+
+  override def executeCommand(commandLine: String,
+                              workflowDescriptor: WorkflowDescriptor,
+                              call: Call,
+                              backendInputs: CallInputs,
+                              scopedLookupFunction: ScopedLookupFunction): Try[Map[String, WdlValue]] = {
+    val callGcsPath = s"${workflowDescriptor.callDir(call)}"
+
+    val engineFunctions = new JesEngineFunctions(GoogleCloudStoragePath(callGcsPath), JesConnection)
+
+    // FIXME: Not particularly robust at the moment.
+    val jesInputs: Seq[JesParameter] = backendInputs.collect({
+      case (k, v) if v.isInstanceOf[WdlFile] => JesInput(k, scopedLookupFunction(k).valueString, Paths.get(v.valueString))
+    }).toSeq
+
+    // Call preevaluateExpressionForFilenames for each of the task output expressions, and flatten the lists into a single Try[Seq[WdlFile]]
+    val filesWithinExpressions = Try(
+      call.task.outputs.map {
+        taskOutput => taskOutput.expression.preevaluateExpressionForFilenames(scopedLookupFunction, engineFunctions)
+      }.flatMap({_.get})
+    )
+
+    val jesOutputs: Try[Seq[JesParameter]] = filesWithinExpressions match {
+      case Failure(error) => Failure(error)
+      case Success(fileSeq) =>
+        val anonOutputs: Seq[JesParameter] = fileSeq map {x => anonymousTaskOutput(x.value, engineFunctions)}
+        // FIXME: If localizeTaskOutputs gives a Failure, need to Fail entire function - don't use .get
+        val namedOutputs: Seq[JesParameter] = taskOutputsToJesOutputs(call.task.outputs, callGcsPath, scopedLookupFunction, engineFunctions).get
+        Success(anonOutputs ++ namedOutputs)
+    }
+
+    // FIXME: Ignore all the errors!
+    val unsafeJesOutputs: Seq[JesParameter] = jesOutputs.get
```

Comment:
While digging through things tnight I realized I have no idea why this is `unsafe` except for that raw `.get`, is that what it refers to?